glossary.md

```diff
@@ -308,6 +310,7 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
 | reinforcement learning            | học tăng cường                            |                                              |
 | representation learning           | học biểu diễn                             | [https://git.io/JvojG](https://git.io/JvojG) |
 | rescale                           | chuyển đổi giá trị                        | [https://git.io/JfeXx](https://git.io/JfeXx) |
+| Residual Network                  | Mạng Phần dư                              |
```

Comment:
Mình thấy không ổn lắm cách dịch này.

Reply:
ok để em nghiên cứu kĩ xem mạng này làm cái gì.

Reply:
```suggestion
| Residual Network                  | Mạng Thặng dư                              |
```
thì sao nhỉ?

Reply:
hmm từ này nghe có vẻ hơi lạ
mạng này có thêm mấy cái skip connections nên dịch là `mạng kết nối dư` được không?

Reply:
Từ `thặng dư` trong `giá trị thặng dư của Marx lênin ấy.

Reply:
hmm ok, mình thấy lạ chắc do chưa học triết :sweat_smile:

Reply:
Triết học đã vào nhà, mở cửa tiễn khách..

Reply:
À... đón khách. Đưa triết vào lý luận liền...

Reply:
hm trong kinh tế thì `thặng dư` có thể dịch từ `surplus` trong `surplus value - giá trị thặng dư`, hoặc từ `residual` trong `residual income - thu nhập thặng dư`, đều nói về phần giá trị dôi ra thêm. Dịch `thặng dư` em thấy cũng ổn về nghĩa, nghe hơi khủng bố thôi 😅

Reply:
Anh Thướng góp ý `residual` nghĩa là `đặc biệt`. Mọi người  thảo luận lại xem sao. 
@ngcthuong Em thấy dùng từ `thặng dư` phản ánh hầu hết các nghĩa của từ đó không riêng trong bối cảnh ResNet. Hơn nữa từ `thặng dư` dịch khá khác biệt, nên như mn thảo luận ở đây thì khi nghe nhiều nó sẽ gắn liền với ResNet mà đỡ bị nhầm sang cái khác. `Mạng đặc biệt` thì từ đặc biệt lại quá chung dẫn đến hơi khó để gợi nhớ nếu nhắc tới nó trong context khác.

Reply:
Từ residual là một từ đặc biệt được sử dụng rất nhiều trong signal processing.  Ngay cả bên ngành mình, từ residual đã được dịch và dùng từ trước rồi, và trong số đó từ "thặng dư" không phản ánh ý nghĩa đúng của từ residual. 

Residual là ám chị sự khác biệt giữa đầu vào và đầu ra, dữ liệu đã qua xử lý dạng x-x_hat. 

Từ "thặng dư" dễ bị hiểu nhầm sang "surplus", mang ý nghĩa "an amount of something left over when requirements have been met; an excess of production or supply over demand." 

Cá nhân mình khi đọc từ "thặng dư" luôn cho mình cảm tưởng sẽ "gain" hoặc có cái gì đó "gain" theo dạng được sinh thêm ra. Nhưng như vậy không phản ánh đúng nghĩa của "residual".

Reply:
Residual function đúng là bằng `F(x) = H(x)-x`, nên F(x) ở đây là phần bị dư ra. 
Trong tiếng anh thì `surplus` hoặc `residual` khác nhau ở ngữ cảnh sử dụng xem mình muốn nhấn mạnh cái gì, surplus sẽ nhấn mạnh `H(x) = F(x) +x`, còn residual sẽ nhấn mạnh ý kia.

Trong tiếng việt cả 2 từ đó đều được dịch là `phần dư, thặng dư` và ko có sự phân biệt tỉ mỉ đó. Giả sử Kaiming He hồi đó có đặt tên là `Surplus Network` thì em nghĩ mình vẫn dịch là `Mạng Thặng dư` thôi.
http://tratu.soha.vn/dict/en_vn/Residual
http://tratu.soha.vn/dict/en_vn/Surplus

Em ko biết từ đó trong SP có cách dịch nào chi tiết hơn ko, chứ thay từ `thặng dư` chỉ vì trùng nghĩa với `surplus` thành `đặc biệt` thì em cũng đồng ý với @davidnvq là ko nên thay.

Reply:
Anh Thướng chia sẻ thế thì em cũng mới biết được thêm nghĩa chuyên ngành của từ `residual`. Nhưng nếu đặt tên là `mạng đặc biệt`, thì rất khó để người đọc muốn tra xem tên tiếng Anh là gì (kiểu specialized, special, v.v.). Nhưng nếu từ `thặng dư`, nghe có thể k thật sự đúng như ý anh mô tả nhưng cũng sẽ dễ dàng hơn để tra ngược lại từ `residual`.

Reply:
@lkhphuc em thấy trong tiếng Việt `phần dư` nhấn mạnh nghĩa `F(x) = H(x) - x`, còn `thặng dư` nhấn mạnh nghĩa `H(x) = F(x) + x` đấy chứ?

Reply:
ơ thì mình nói vậy mà @cuongvng

Reply:
Nếu vậy thì `phần dư` sẽ tương đương `residual`, còn `thặng dư` sẽ ứng với `surplus` mà anh @lkhphuc. Vậy thì nên dịch là `Mạng Phần dư` chứ?
> Residual function đúng là bằng F(x) = H(x)-x, nên F(x) ở đây là phần bị dư ra.
Trong tiếng anh thì surplus hoặc residual khác nhau ở ngữ cảnh sử dụng xem mình muốn nhấn mạnh cái gì, surplus sẽ nhấn mạnh H(x) = F(x) +x, còn residual sẽ nhấn mạnh ý kia.

Reply:
À chỗ này thì mình ko có ý kiến, mình hiểu `thặng dư` và `phần dư` là như nhau, trong 2 link mình đưa ở trên thì cả 2 cách dịch đều có trong cả 2 link.

Reply:
Trong hình `../img/residual-block.svg` thì `F(x) +x` được nhấn mạnh hơn nên ý kiến của em là giữ nguyên cách dịch `thặng dư`.

Reply:
> @lkhphuc em thấy trong tiếng Việt `phần dư` nhấn mạnh nghĩa `F(x) = H(x) - x`, còn `thặng dư` nhấn mạnh nghĩa `H(x) = F(x) + x` đấy chứ?

Hai cái đó giống nhau, chỉ khác dấu thôi em. Không có ý nghĩa khác biệt.

Reply:
Như mình nói từ "residual" ở mạng này không có nghĩa đặc biệt nào cả, nó cùng nghĩa với "residual" thông thường sử dụng trong tất cả các mảng khác. 

Cái từ mình không ổn là từ "thặng" vì, tự nhiên mình lại đưa ra một nghĩa dịch khác cho từ "residual" mà theo mình ý dịch là không cần thiết phải khác biệt như vậy. 

Trong xử lý tín hiệu, đặc biệt quá trình nén, phần "residual" của "predictor" và "original" frames/images gọi là "residual image". Trong dó predictor là ảnh/thông tin có thể dự đoán được, đã biết trước, và phần residual image sẽ được nén để gửi thêm sang bên thu. Đây là ảnh residual chứa phần thông tin "dư thừa", cần bổ xung, chứ không phải thông tin "thặng dư".

Reply:
> Trong hình `../img/residual-block.svg` thì `F(x) +x` được nhấn mạnh hơn nên ý kiến của em là giữ nguyên cách dịch `thặng dư`.

Em thử huấn luyện mạng neural network thay + = -, mình tin kết quả sẽ giống nhau.

Reply:
@ngcthuong em đồng ý với anh là khi đọc `thặng dư` sẽ cho cảm giác có gì đó được sinh thêm ra. Trong hình và code có thể thấy là đầu vào `x` đi qua `residual block` thì được *cộng thêm* giá trị f(x) giống như regular conv block. Như vậy em nghĩ dịch là `thặng dư` sẽ ổn hơn khi nhấn mạnh phần f(x) được cộng thêm này.
> Cá nhân mình khi đọc từ "thặng dư" luôn cho mình cảm tưởng sẽ "gain" hoặc có cái gì đó "gain" theo dạng được sinh thêm ra. Nhưng như vậy không phản ánh đúng nghĩa của "residual".

Còn việc huấn luyện mạng neural network thay + = - thì em không nghĩ kết quả sẽ giống nhau, vì sau khi cộng thêm (hoặc trừ đi) sẽ phải qua ReLU. Rõ ràng thay + = - sẽ có thể làm đổi dấu kết quả, và qua ReLU sẽ cho kích hoạt khác nhau.

Reply:
Cái cộng trừ thì a Thướng nói chuẩn đó Cường. Khi kì vọng cùng một đầu ra thì ở giây phút cuối cùng optim sẽ tối ưu weight của trường hợp (-) ngược dấu với trường hợp (+) tuy gradient và weight trong lúc học có thể khác nhau và thời gian học khác nhau. Mình cũng thử một vài lần và kq giữa (+ và -), (* và /).

Reply:
@davidnvq tức là mọi người đang nói đến kết quả đầu ra cuối cùng của mạng? Nếu vậy thì em không có ý kiến gì.

Reply:
bên #1411 @davidnvq  kêu đang chờ thảo luận bên này mà bên này mình ko thấy có action gì để chờ cả, nên làm cái tổng kết nhé:

* Residual chỉ `F(x) = H(x) - x` và Surplus chỉ `H(x) = F(x) + x` , 2 mặt của một đồng xu.
* Residual và Surplus dịch ra tiếng việt đều là `phần dư, phần thừa, thặng dư`, theo các từ điển mình tra: http://tratu.soha.vn/dict/en_vn/Residual , http://tratu.soha.vn/dict/en_vn/Surplus
* @cuongvng cho ví dụ: `surplus value: giá trị thặng dư` và `residual income: nhu nhập thặng dư` https://vietnambiz.vn/thu-nhap-thang-du-residual-income-la-gi-20200319153226272.htm 
* `Mạng Phần dư` nghe không ổn nên mọi người thống nhất mạng `Mạng Thặng dư`.
* @ngcthuong kêu `thặng dư` chỉ mang ý nghĩa `surplus` thôi.
* @cuongvng kêu cho dù người đọc có hiểu `thặng dư` là `surplus` thì vẫn chính xác trong trường hợp resnet, vì trong paper cũng như trong diagram họ sử dụng kí hiệu `H(x) = F(x) + x` khá nhiều. (mình đồng ý)

Vì chưa ai có đề xuất gì thay chữ `Thặng dư` cả nên mình quay lại max pooling với các option cũ:
* 👍 Mạng Thặng dư
* 🎉 Mạng Phần dư
* 👀 Mạng Phần thừa

Kêu gọi cử tri: @duythanhvn @minhduc0711 @cuongvng @ngcthuong @davidnvq @tiepvupsu @quangnhat185 @rootonchair ..

Reply:
Mình cần xác định rõ ở đây là "dấu +/-" không gây ra khác biệt ở kết quả cuối cùng. Định nghĩa residual là sự khác biệt giữa 2 tín hiệu. A-B, hay B-A thì vẫn là một thôi. 

Mình không nghĩ lấy ví dụ bên kinh tế là phụ hợp, chưa kể mình cũng không chắc họ dịch có sát nghĩa không. Điển hình như BKAV suốt ngày dịch computational photography là "nhiếp ảnh điện toán" nhưng theo mình cách dịch đó là sai. 

Thặng dư mang nhiều "hướng" về surplus hơn hơn là residual. Trong ngôn ngữ rất khó phân biệt hoàn toàn dạng như thế này. 

Xét về mặt chung, thì có thể chấp nhận được dùng "thặng dư" cho cấu trúc này, nhưng hoàn toàn cần phải xem lại với các từ khóa khác liên quan tới residual. 

P/s: Mình xin lỗi vì miss phần thảo luận lần trước. Nhưng từ khóa cần phải chuẩn, không chuẩn thì sau rất khó sửa.

Reply:
ca này khó quá nhỉ, mình cũng chưa hiểu thực sự bản chất của ResNet nên đành vote trắng vậy

Reply:
Nhờ anh @tiepvupsu phản hồi thêm ý kiến cho phần thảo luận này ạ.

Reply:
Mình từng dùng cả dense và residual trong bài báo của mình, cũng từng mô hình lại cấu trúc nested residual (khác với densely residual network), về mặt cơ bản thì là nó giống nhau thôi. 

Mình nhận thấy dùng dấu + trong bài residual là để thuận hơn là dấu trừ (giống như mình nói cộng-trừ nhân - chia). Nên phép cộng và nhân được ưu tiên viết tên thành hàm khi có chung ý nghĩa. 

Còn với mạng network khi bộ thông số là tự học ra, thì nếu đặt giá trị + hoặc - ngay từ đầu thì các hệ số của weight cũng sẽ thay đổi tương ứng, chứ không phải do ảnh hưởng của ReLu mà kết quả khác biệt. Dĩ nhiên cái này là suy đoán logic, để xác thực cần thử nghiệm nhiều lần. Việc thay đổi giá trị cũng có thể dẫn đến làm lệch các bước nhảy gradient, hay điểm xuất phát. Nhưng mình nghĩ không khác biệt nhiều.

Reply:
Xin phép anh em đào lại vấn đề vì có lẽ chưa thoả mãn. Mình và Đức bỏ phiếu trắng chỗ này vì không quá chắc về sự khác biệt.

Hiện tại mình cũng chưa thấy những từ khóa khác liên quan tới residual, tuy nhiên cũng muốn rộng đường thảo luận trong tương lai nếu phát sinh thì mình thấy anh em stuck ở hai nhóm từ `phần dư` và `thặng dư`. `thặng dư` thì anh em đã trao đổi rất chi tiết rồi, vậy thì mình muốn hỏi thêm ở chỗ Vinh với Phúc là từ `phần dư` này anh em thấy chưa ổn cụ thể là chưa ổn như thế nào vậy nhỉ?

Reply:
Các từ khóa khác liên quan tới "residual"
- Residual learning
- Residual signal: the residual of an observed value is the difference between the observed value and the estimated value of the quantity of interest (https://en.wikipedia.org/wiki/Errors_and_residuals)
- Residual analysis: Residuals are differences between the one-step-predicted output from the model and the measured output from the validation data set. Thus, residuals represent the portion of the validation data not explained by the model. (https://www.mathworks.com/help/ident/ug/what-is-residual-analysis.html)
- Residual image/residual coding/

Reply:
Như mọi người thảo luận ở trên, chúng ta có lẽ ai cũng đã hiểu từ "Residual" là sự khác biệt giữa cái thực tế và cái dự đoán. Phần khác biệt này có thể dương hoặc âm.

Về "Phần dư" và "Thặng dư" thì mình thấy có chút khác biệt ở chỗ "Thặng" mang nặng nghĩa "dương" còn "phần" thì mang nghĩ trung tính, mặc dù từ "dư" cũng có nghĩa là "dương" trong tiếng Việt.

Mình nghĩ trong tiếng Việt cũng có những chỗ một từ thường mang nghĩa dương tính nhưng vẫn được đặt trong những bối cảnh mà nó có thể mang nghĩa âm. "*Lợi* nhuận" vẫn có thể âm, "*Tăng* trưởng" vẫn có thể âm, và thực ra ngay cả "*thặng dư*" [cũng vẫn có thể âm](https://www.youtube.com/watch?v=eSKsQwBsjlU). Bonus thêm là [cũng có "negative surplus"](https://www.sciencedirect.com/science/article/abs/pii/S0167668797000140) luôn.

Vì vậy, cá nhân mình  nghĩ chúng ta không nên mất thời gian tranh luận chỗ "thặng" là surplus  và chỉ có thể dương. Tuy nhiên, nếu phải chọn giữa "Thặng dư" và "phần dư" thì mình chọn là "Phần dư" vì nó mang nghĩa trung tính hơn. "Thặng" xem ra là một thứ mà chúng ta muốn đạt được, ngược với tư tưởng của "Residual" trong CS và EE _thường_ là những thứ được mong muốn bằng 0.

Reply:
Like "tư tưởng của "Residual" trong CS và EE thường là những thứ được mong muốn bằng 0"

Reply:
Vì mọi người có vẻ đồng tình.Vậy em xin chốt là `residual` mình sẽ dịch là `phần dư`. Và `residual network` sẽ là `mạng phần dư`.